### PR TITLE
libgit: add loose object pruning to GC

### PR DESCRIPTION
**Pull request by strib**

### CODE REVIEW

File: libgit/repo.go [hunk 1]
```diff
@@ -846,17 +875,37 @@ func GCRepo(
...
-	log.CDebugf(ctx, "Packing %s loose refs", numLooseRefs)
-	return storer.PackRefs()
+	if doPackRefs {
+		log.CDebugf(ctx, "Packing %s loose refs", numLooseRefs)
+		err = storage.PackRefs()
+		if err != nil {
+			return err
+		}
+	}
+
```
Comment (taruti) [inline libgit/repo.go]: Wouldn't it be better to first prune and then pack. And not pack and then prune. (if prune deletes some objects then adding them to a pack is not good). I.e. just change the order of pruning and packing here?

File: libgit/repo.go [hunk 2]
```diff
@@ -846,17 +875,37 @@ func GCRepo(
...
+	if !doPackRefs && !doPruneLoose {
 		log.CDebugf(ctx, "GC no longer needed")
 		return nil
 	}
 
-	// For now just pack refs.  TODO: add object pruning and object
-	// re-packing.
-	log.CDebugf(ctx, "Packing %s loose refs", numLooseRefs)
-	return storer.PackRefs()
+	if doPackRefs {
```
Comment (taruti) [inline libgit/repo.go]: Wouldn't it be better to first prune loose objects and then pack. I.e. change the order of packing and pruning here.

File: libgit/repo.go (hunk 1 above)
Comment (strib) [reply libgit/repo.go]: Above is just packing refs, not objects.  It won't ever delete or unreference any objects.  I don't think the order matters, but please correct me if I'm wrong.

Comment (taruti) [reply libgit/repo.go]: Sorry for the noice, you are right.

Comment (taruti) [reply libgit/repo.go]: Sorry for the noise, you are right.

**Pull request merged**